docs/api/modulus.models.rst

```diff
@@ -2,6 +2,403 @@
+        start_time = time.time()
+        loss = training_step(input, output)
+        elapsed_time = time.time() - start_time
+        print(f"Iteration {i + 1} took {elapsed_time:.4f} seconds")
+        input.copy_(torch.randn(8, 1, 128, 128).to("cuda"))
+
+For the simple model above, you can observe ~1.1x speed-up due to CUDA Graphs and AMP.
+The speed-up observed changes from model to model and is typically greater for more
+complex models. 
+
+.. note::
+    The ``ModelMetaData`` and ``modulus.model.module.Module`` do not make the model
+    support CUDA Graphs, AMP, etc. optimizations automatically. The user is responsible
+    to write the model code that enables each of these optimizations. 
+
+.. note::
+    The ``StaticCaptureTraining`` decorator is still under development and may be
+    refactored in the future.
+
+
+Converting PyTorch Models to Modulus Models
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the above example we show constructing a Modulus model from scratch. However you
+can also convert existing PyTorch models to Modulus models in order to leverage
+Modulus features. To do this, you can use the ``Module.from_torch`` method as shown
+below.
+
+.. code:: python
+
+    from dataclasses import dataclass
+    from modulus.models.meta import ModelMetaData
+    from modulus.models.module import Module 
+    import torch.nn as nn
+
+    class TorchModel(nn.Module):
+        def __init__(self):
+            super(TorchModel, self).__init__()
+            self.conv1 = nn.Conv2d(1, 20, 5)
+            self.conv2 = nn.Conv2d(20, 20, 5)
+    
+        def forward(self, x):
+            x = self.conv1(x)
+            return self.conv2(x)
+
+    @dataclass
+    class MetaData(ModelMetaData):
+        name: str = "UNet"
+        # Optimization
+        jit: bool = False
+        cuda_graphs: bool = True
+        amp_cpu: bool = True
+        amp_gpu: bool = True
+
+    ModulusModel = Module.from_torch(TorchModel, meta=MetaData())
+
+
+Saving and Loading Modulus Models
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As mentioned above, Modulus models are interoperable with PyTorch models. This means that
+you can save and load Modulus models using the standard PyTorch APIs however, we provide
+a few additional utilities to make this process easier. A key challenge in saving and
+loading models is keeping track of the model metadata such as layer sizes, etc. Modulus
+models can be saved with this metadata to a custom file ``.mdlus``. These files allow
+for easy loading and instantiation of the model. Below is an example of saving and
+instantiating a model from a ``.mdlus`` file.
+
+.. code:: python
+
+    >>> import torch
+    >>> from modulus.models import Module
+    >>> from modulus.models.mlp.fully_connected import FullyConnected
+    >>> model = FullyConnected(in_features=32, out_features=64)
+    >>> model.save("model.mdlus") # Save model to .mdlus file
+    >>> model.load("model.mdlus") # Load model weights from .mdlus file
```

Comment:
Add comment saying that here we already have the object instantiated with the right input parameters.